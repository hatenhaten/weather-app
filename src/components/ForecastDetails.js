// imports

import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import WeatherIcon from "react-icons-weather";

// component

// prettier-ignore
const ForecastDetails = ({ forecast }) => {
    const { date, icon, temperature, humidity, wind } = forecast;

    return (
        <div className="forecast-details" data-testid="forecast-details">
            <div className="forecast-details__date">
                {moment(date).format("ddd Do MMM")}
            </div>
            <div className="forecast-details__icon" data-testid="forecast-icon">
                <WeatherIcon name="owm" iconId={icon} />
            </div>
            <div className="forecast-details__max-temp">
                <p>
                    Max Temperature: <span>{temperature.max}&deg;C</span>
                </p>
            </div>
            <div className="forecast-details__min-temp">
                <p>
                    Min Temperature: <span>{temperature.min}&deg;C</span>
                </p>
            </div>
            <div className="forecast-details__humidity">
                <p>
                    Humidity: <span>{humidity}%</span>
                </p>
            </div>
            <div className="forecast-details__wind">
                <p>
                    wind: <span>{wind.speed}mph {wind.direction}</span>
                </p>
            </div>
        </div>
    );
};

// prop-validation

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    description: PropTypes.string,
    icon: PropTypes.string,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }),
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
    humidity: PropTypes.number,
  }).isRequired,
};

// exports

export default ForecastDetails;
