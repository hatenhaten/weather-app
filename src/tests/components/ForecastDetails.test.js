// imports

import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

// snapshot tests

describe("ForecastDetails", () => {
  const validProps = {
    forecast: {
      date: 1525046400000,
      icon: "800",
      temperature: {
        min: 12,
        max: 22,
      },
      wind: {
        speed: 13,
        direction: "s",
      },
      humidity: 30,
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails forecast={validProps.forecast} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  // value tests

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastDetails forecast={validProps.forecast} />
    );

    expect(getByText("Mon 30th Apr")).toHaveClass("forecast-details__date");
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-details__icon");
  });
});
