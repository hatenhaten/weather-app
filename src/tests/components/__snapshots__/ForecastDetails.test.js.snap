// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Mon 30th Apr
    </div>
    <div
      class="forecast-details__icon"
      data-testid="forecast-icon"
    >
      <i
        class="wi wi-day-sunny"
        name="owm"
      />
    </div>
    <div
      class="forecast-details__max-temp"
    >
      <p>
        Max Temperature: 
        <span>
          22°C
        </span>
      </p>
    </div>
    <div
      class="forecast-details__min-temp"
    >
      <p>
        Min Temperature: 
        <span>
          12°C
        </span>
      </p>
    </div>
    <div
      class="forecast-details__humidity"
    >
      <p>
        Humidity: 
        <span>
          30%
        </span>
      </p>
    </div>
    <div
      class="forecast-details__wind"
    >
      <p>
        wind: 
        <span>
          13mph s
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
